src/Composer/AddressSplitter.php

```diff
@@ -38,12 +38,23 @@ public function split(array $addresses): array
             $uri = $pathParts[0];
             $unitId = (int)$pathParts[1];
             // sort by address and size to help chunking
+            // for bytes address type with same address: first byte, second byte
             usort($addrs, function (Address $a, Address $b) {
                 $aAddr = $a->getAddress();
                 $bAddr = $b->getAddress();
                 if ($aAddr === $bAddr) {
                     $sizeCmp = $a->getSize() <=> $b->getSize();
-                    return $sizeCmp !== 0 ? $sizeCmp : $a->getType() <=> $b->getType();
+                    if ($sizeCmp !== 0) {
+                        return $sizeCmp;
+                    }
+                    $typeCmp = $a->getType() <=> $b->getType();
+                    if ($typeCmp !== 0) {
+                        return $typeCmp;
+                    }
+                    if ($a->getType() === Address::TYPE_BYTE) {
+                        return $b->isFirstByte();
```

Comment:
PHPstan check `vendor/bin/phpstan analyse` does not know what type `$b` is.  

This change would makes check pass
```php
                if ($a instanceof ByteReadRegisterAddress && $b instanceof ByteReadRegisterAddress) {
                    return $b->isFirstByte();
                }
```